Cme unicode issue hotseat

caseylitton opened the pull request:
@stvstnfrd @dcadams @kluo @caesar2164 PR for fixing non ascii student name hot seat issue for the cme renderer.  Fixed associated test.

Creating certs with non-ascii student names like 'Raquel Sá Lopes da Slva' had two problems.
1.  Could not save pdfs for any cert locally because the student name with non-ascii chars was used in the file name.  I couldn't find a good way to do this, the code just replaces 'bad' chars with a dash
1.  CME certs would fail because there is a second formatting done on the student name because of the designations.  That was a decoding issue with a string format.

caesar2164 commented on gen_cert.py:
```diff
@@ -796,9 +796,9 @@ def _generate_v4_certificate(
                 # Add student name designation if not Other or None
                 if designation not in ['Other', 'None']:
-                    student_name = u"{name}, {designation}".format(
+                    student_name = "{name}, {designation}".format(
```
this seems bad, I think the prepended "u" means "unicode"...

caesar2164 commented on create_pdfs.py:
```diff
@@ -116,6 +116,8 @@ def main():
             certificate_data.append((name, course, args.long_org, args.long_course, download_url))
             gen_dir = os.path.join(cert.dir_prefix, S3_CERT_PATH, download_uuid)
+            # Remove non-ascii chars since this is a filename
+            name = ''.join([i if ord(i) < 128 else ' ' for i in name])
```
clearly this works, but don't really like it...

caesar2164 commented on gen_cert.py:
```diff
@@ -796,9 +796,9 @@ def _generate_v4_certificate(
+                    student_name = "{name}, {designation}".format(
                         name=student_name,
-                        designation=designation.decode('utf-8'),
+                        designation=designation,
```
yes, agreed!

caseylitton committed:
Let non-ascii student names save certs locally

Cert PDFs are saved locally in non-production
environments with the student name added to the
filename.  If the student name has non-ascii
characters then certficate creation would break.

caseylitton committed:
Fix encoding mismatches when adding title to name

stvstnfrd commented on gen_cert.py:
```diff
@@ -796,12 +796,12 @@ def _generate_v4_certificate(
                     )
                 achievements_string = value['credits']
-                achievements_description_string = self.cert_data['CREDITS']
+                achievements_description_string = unicode(self.cert_data['CREDITS'])
```
This is the extra change that was needed? @caseylitton

caseylitton replied:
@stvstnfrd yes, the achievements designation was previously unicode, so I made the "new" designation code follow suit.
